### PR TITLE
Cleanup default features, remove dnssec from client and server default features

### PR DESCRIPTION
**Pull request by bluejekyll**
This is a series of 5 commits to build up the the final commit to remove the default `dnssec` feature from the `client` crate.

### CODE REVIEW

File: crates/server/src/authority/auth_lookup.rs [hunk 1]
```diff
@@ -363,17 +355,37 @@ impl<'a> IntoIterator for &'a LookupRecords {
         match self {
             LookupRecords::Empty => LookupRecordsIter::Empty,
             LookupRecords::Records {
-                is_secure,
-                supported_algorithms,
+                lookup_options,
                 records,
-            } => LookupRecordsIter::RecordsIter(records.records(*is_secure, *supported_algorithms)),
-            LookupRecords::ManyRecords(is_secure, supported_algorithms, r) => {
```
Comment (djc) [inline crates/server/src/authority/auth_lookup.rs:370]: There's quite a bit of repetition here. Can we extract the `LookupRecordsIter::ManyRecordsIter` out of the `cfg_if!()` call, and maybe also add a convenience method (either as method to `LookupOptions` or a local helper function here) that can support the different scenarios here?

File: crates/server/src/authority/authority.rs [hunk 2]
```diff
@@ -11,11 +11,78 @@ use std::pin::Pin;
...
+/// For example, `is_dnssec` will include `RRSIG` in the response, `supported_algorithms` will only include a subset of
+///    `RRSIG` based on the algorithms supported by the request.
+#[derive(Clone, Copy, Debug)]
+pub struct LookupOptions {
+    is_dnssec: bool,
+    #[cfg(feature = "dnssec")]
+    supported_algorithms: SupportedAlgorithms,
+}
+
+impl Default for LookupOptions {
```
Comment (djc) [inline crates/server/src/authority/authority.rs:34]: This can be derived.

**Review by djc — COMMENTED** [review]
I think this generally looks good!

It does feel like you're maybe overusing `cfg_if!` when you can also just do `if cfg!(feature = "dnssec") { }`. The latter gets evaluated at runtime, but unless there are structural differences introduced by the option, it seems like it should work in most cases, and it saves on an extra level of nesting.

File: crates/server/src/authority/auth_lookup.rs (hunk 1 above)
Comment (bluejekyll) [reply crates/server/src/authority/auth_lookup.rs:370]: Thanks for that feedback. Let me see about deduping this.

Comment (bluejekyll) [reply crates/server/src/authority/auth_lookup.rs:370]: Ok, great suggestion. I think this is much cleaner now.

**Comment by bluejekyll** [comment]
I think this is ready now...

**Review by djc — APPROVED** [review]
Skimmed over this, seems like a nice improvement!

**Pull request merged**